mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/parsing/nativequery/NativeQueryParserTest.java

```diff
@@ -823,5 +827,142 @@ public void shouldSupportDotInCollectionName() {
+		}
+	}
+
+	@Test
+	public void shouldRecognizeMissingCollectionName() {
+		try {
+			NativeQueryParser parser = Parboiled.createParser( NativeQueryParser.class );
+			ParsingResult<MongoDBQueryDescriptorBuilder> run = new BasicParseRunner<MongoDBQueryDescriptorBuilder>( parser.Query() )
+				.run( "db.find( { 'name': 'Andy' } )" );
+
+			run.resultValue.build();
+			Assert.fail( "An exception was expected" );
+		}
+		catch (Exception e) {
+			assertThat( e ).isInstanceOf( NativeQueryParseException.class ).hasMessage( "Cli query should match the format db.collection.oper()" );
+		}
+	}
+
+	@Test
+	public void shouldRecognizeMissingBrackets() {
+		try {
+			NativeQueryParser parser = Parboiled.createParser( NativeQueryParser.class );
+			ParsingResult<MongoDBQueryDescriptorBuilder> run = new BasicParseRunner<MongoDBQueryDescriptorBuilder>( parser.Query() )
+					.run( "db.Person.find( { 'name': 'Andy' } " );
 
+			run.resultValue.build();
+			Assert.fail( "An exception was expected" );
+		}
+		catch (Exception e) {
+			assertThat( e ).isInstanceOf( NativeQueryParseException.class ).hasMessage( "Missing ')'" );
+		}
+	}
+
+	@Test
+	public void shouldRecognizeInvalidJSON() {
+		try {
+			NativeQueryParser parser = Parboiled.createParser( NativeQueryParser.class );
+			ParsingResult<MongoDBQueryDescriptorBuilder> run = new BasicParseRunner<MongoDBQueryDescriptorBuilder>( parser.Query() )
+					.run( "{ \"foo\" true }" );
+
+			run.resultValue.build();
+			Assert.fail( "An exception was expected" );
+		}
+		catch (Exception e) {
+			assertThat( e ).isInstanceOf( NativeQueryParseException.class );
+			assertThat( e.getCause() ).isInstanceOf( JsonParseException.class );
+		}
+	}
+
+	@Test
+	public void shouldRecognizeInvalidJSONAsParameter() {
+		try {
+			NativeQueryParser parser = Parboiled.createParser( NativeQueryParser.class );
+			ParsingResult<MongoDBQueryDescriptorBuilder> run = new BasicParseRunner<MongoDBQueryDescriptorBuilder>( parser.Query() )
+					.run( "db.Person.find( { 'name: 'Andy' } ) " );
+
+			run.resultValue.build();
+			Assert.fail( "An exception was expected" );
+		}
+		catch (Exception e) {
+			assertThat( e ).isInstanceOf( NativeQueryParseException.class );
+			assertThat( e.getCause() ).isInstanceOf( JsonParseException.class );
+		}
+	}
+
+	@Test
+	public void shouldRecognizeInvalidJSONArrayAsParameter() {
+		try {
+			NativeQueryParser parser = Parboiled.createParser( NativeQueryParser.class );
+			ParsingResult<MongoDBQueryDescriptorBuilder> run = new ReportingParseRunner<MongoDBQueryDescriptorBuilder>( parser.Query() )
+					.run( "db.Order.aggregate([  ])" );
+
+			run.resultValue.build();
+			Assert.fail( "An exception was expected" );
+		}
+		catch (Exception e) {
+			assertThat( e ).isInstanceOf( NativeQueryParseException.class ).hasMessage( "Parameters are invalid for aggregate" );
+		}
+	}
+
+	@Test
+	public void shouldRecognizeMissingDBPart() { // ????
```

Comment:
Why the "// ????" ?

Reply:
I just marked the line and forgot to remove it :( Actually I thought the thrown error would be confusing for a user because this query looks very much like the ordinary query but it says it's a JSON parse error. I wonder if there is some clever way to identify this but maybe I'm overthinking this.

Reply:
I would say let's make the error message meaningful and that should be enough.